ark/docs/components/AutoDocs.tsx

```diff
@@ -0,0 +1,67 @@
+import { flatMorph } from "@ark/util"
+import { ark } from "arktype"
+import { Fragment } from "react"
+
+export const AutoDocs = () => {
+	const flattened = flatMorph(ark.internal.resolutions, (k, v) =>
+		"description" in v ? [k, v.description] : []
+	)
+
+	const groupedEntries = Object.entries(flattened).reduce(
```

Comment:
Nit: I would probably write this as a single `flatMorph` call, where `root` is defined ahead of time as an empty array that you push to directly in those cases and other just ends up getting created out of the entries.

Reply:
I don't think I understand `flatMorph` well enough to know what you mean here. 

I do see how this logic could be done in a single loop wrt the logic of it, but I don't know how I'd write it with `flatMorph`

Reply:
Okay no problem I can do that and I will ping you if it ends up actually being a good idea :pray: